src/hooks/useUSDCPrice.ts

```diff
@@ -24,7 +24,7 @@ export const STABLECOIN_AMOUNT_OUT: { [chainId: number]: CurrencyAmount<Token> }
 export default function useUSDCPrice(currency?: Currency): Price<Currency, Token> | undefined {
   const chainId = currency?.chainId
 
-  const amountOut = chainId ? STABLECOIN_AMOUNT_OUT[chainId] : undefined
+  const amountOut = currency && chainId ? STABLECOIN_AMOUNT_OUT[chainId] : undefined
```

Comment:
does this actually prevent any work? Won't `v2USDCTrade` and `v3USDCTrade` skip the query when `currency` is undefined, and `useMemo` has an if check on `currency` early